src/lib/amount.ts

```diff
@@ -147,6 +147,27 @@ export function getPoolRatioAmount(
   return ratio;
 }
 
+export function getSlippageAmount(
+  amount: string,
+  wei: number,
+  slippage: string,
+  increase: boolean
+): string {
+  let precision = 30;
```

Comment:
for this 30 precision, it is using in both UI and ts-sdk. I think better use a constant in ts-sdk and in UI, use the constant so in the future if we change it, we do not need find different places to change it.